File: backend/src/main/java/com/cdac/MovieBooking/Service/AdminService.java
package com.cdac.MovieBooking.Service;

import com.cdac.MovieBooking.Dtos.Request.MovieDTO;
import com.cdac.MovieBooking.Dtos.Response.AdminDashboardStatsDTO;
import com.cdac.MovieBooking.Entities.Movie;
import com.cdac.MovieBooking.Entities.Theatre;
import com.cdac.MovieBooking.Entities.User;

import java.util.List;

public interface AdminService {
    AdminDashboardStatsDTO getDashboardStats();

    List<User> getAllUsers();

    User updateUserStatus(Long userId);

    List<Movie> getAllMovies();

    Movie addMovie(MovieDTO movieDTO);

    Movie updateMovie(Long id, MovieDTO movieDTO);

    Movie updateMovieStatus(Long id);

    List<Theatre> getPendingTheatres();

    Theatre approveTheatre(Long theatreId, boolean isApproved);
}

File: frontend/src/services/adminService.js
import axiosInstance from "./axiosInstance";

// Stats
export const getDashboardStats = () => {
    return axiosInstance.get("/admin/stats");
};

// Users
export const getAllUsers = () => {
    return axiosInstance.get("/admin/users");
};

export const updateUserStatus = (userId) => {
    return axiosInstance.patch(`/admin/users/${userId}/status`);
};

// Movies
export const getAllMovies = () => {
    return axiosInstance.get("/admin/movies");
};

export const addMovie = (movieData) => {
    return axiosInstance.post("/admin/movies", movieData);
};

export const updateMovie = (id, movieData) => {
    return axiosInstance.put(`/admin/movies/${id}`, movieData);
};

export const updateMovieStatus = (id) => {
    return axiosInstance.patch(`/admin/movies/${id}/status`);
};

// Approvals
export const getAllTheatres = () => {
    return axiosInstance.get("/admin/theatres");
};

export const getPendingTheatres = () => {
    return axiosInstance.get("/admin/approvals");
};

export const approveTheatre = (id, approved) => {
    return axiosInstance.patch(`/admin/approvals/${id}?approved=${approved}`);
};
